gems/sorbet-runtime/lib/types/utils.rb

```diff
@@ -79,10 +79,12 @@ def self.unwrap_nilable(type)
     case type
     when T::Types::Union
       non_nil_types = type.types.reject {|t| t == Nilable::NIL_TYPE}
-      if non_nil_types.length == 1
-        non_nil_types.first
+      return nil if type.types.length == non_nil_types.length
+      case non_nil_types.length
+      when 0 then nil
+      when 1 then non_nil_types.first
       else
-        nil
+        T::Types::Union::Private::Pool.union_of_types(non_nil_types[0], non_nil_types[1], non_nil_types[2..])
```

Comment:
@jez this will break for Ruby <2.5

Reply:
Thanks. We only support 2.6 and above

https://github.com/sorbet/sorbet/pull/4030/files#diff-9cf20d6cd4e2fe34e9089339cf947f3c6c455d58b2a75e5d84f843b83fcbd239L255-R257

Reply:
I read this as untested, not explicitly unsupported.

We can change this and still keep compatibility with Ruby 2.5: https://github.com/sorbet/sorbet/pull/4261

Reply:
@jez consider updating the gemspecs (filed as https://github.com/sorbet/sorbet/issues/4265 before i thought to look here)